### PR TITLE
Update A Ton Of Dependencies To Fix Mac Releases

### PR DESCRIPTION
**Pull request by X-sam**
This should make Mac builds work.

This PR brings the following changes:
- Update our base Python dependency to 3.11, the latest stable release at the time of this PR.
- Update our Qt dependency from Qt5 to Qt6, migrating us from `PyQt5` to `PySide6`. (as PySide6 provides universal binaries and PyQt does not)
- Update our `Pillow` dependency to 10.0 (needed for Mac universal binary support)
- Lock our `numpy` & `requests` dependencies to specific versions (to ensure build consistency).
- Break out build and development dependencies into `requirements.build.txt` & `requirements.txt` (so as not to pollute the build instance with e.g. `pre-commit`)
- Update README.md to reflect actual build requirements on all supported platforms, as well as some adjustments to clean up the flow and reduce repetition.
- Update our Github Actions builder to use the new Apple m1 runner for mac builds- this greatly simplifies building universal binaries for mac.

Additionally, this change also required some hacking on fbs, which I've similarly upgraded from Qt5 to Qt6, covered in this PR: AllYarnsAreBeautiful/fbs#6

### CODE REVIEW

**Review by t0mpr1c3 — APPROVED** [review]

**Comment by t0mpr1c3** [comment]
Are there corresponding changes needed to the AYAB fork of fbs?

**Comment by X-sam** [comment]
> Are there corresponding changes needed to the AYAB fork of fbs?

linked in the description ;P

**Comment by t0mpr1c3** [comment]
> > Are there corresponding changes needed to the AYAB fork of fbs?
> 
> linked in the description ;P

Sorry, was following on my phone and didn't see that, or that this was a draft PR.

File: src/main/python/ayab/signal_receiver.py
Comment (X-sam) [inline src/main/python/ayab/signal_receiver.py:1]: @t0mpr1c3 Can you explain what the `signals` function is supposed to do? I think it needs to be changed to match the `PySide` Signal implementation, but I don't totally follow where our `SignalReceiver.signals()` is used...
https://github.com/AllYarnsAreBeautiful/ayab-desktop/pull/572/files#diff-92d3ed790822f98ffc64780bfae9485fbf3994ecf24be4f693fe402e3e04a10fL64

Comment (X-sam) [reply src/main/python/ayab/signal_receiver.py:1]: turns out it was `SignalInstance`. Updated in 9ef2f45

File: requirements.build.txt
Comment (dl1com) [inline requirements.build.txt:1]: Before merging this PR, we have to change this to the tag/commit which results out of this PR: https://github.com/AllYarnsAreBeautiful/fbs/pull/6

File: src/main/python/ayab/engine/communication.py
Comment (dl1com) [inline src/main/python/ayab/engine/communication.py:121]: Why did you remove the disable_hardware_beep setting throughout this PR?
The API.md still shows it as a valid parameter (https://github.com/AllYarnsAreBeautiful/fbs/pull/6).

**Comment by dl1com** [comment]
Looks very promising!
Let's wait for @t0mpr1c3 's review on this and the fbs PR before proceeding.

File: src/main/python/ayab/engine/options_gui.ui
Comment (t0mpr1c3) [inline src/main/python/ayab/engine/options_gui.ui:247]: Possible regression?

File: src/main/python/ayab/engine/options_gui.ui
Comment (t0mpr1c3) [inline src/main/python/ayab/engine/options_gui.ui:266]: Possible regression?

File: src/main/python/ayab/engine/options_gui.ui
Comment (X-sam) [reply src/main/python/ayab/engine/options_gui.ui:247]: Resources are handled differently- if you load the app you'll see this still renders correctly.

File: src/main/python/ayab/signal_receiver.py
Comment (t0mpr1c3) [reply src/main/python/ayab/signal_receiver.py:1]: I can't remember if `SignalReceiver.signals()` is used anywhere. Does it throw an error if you delete it?

**Comment by X-sam** [comment]
Yeah it's important it's part of how they get registered apparently.

**Comment by t0mpr1c3** [comment]
Do you want me to go through these other potential regressions? I think it should be apparent from looking at the blame what needs to be reinstated, which I haven't done yet, so I can do that if that would save you some time.

**Comment by X-sam** [comment]
I'm trying to figure out what happened, must've been a bad merge commit?

**Comment by t0mpr1c3** [comment]
I expect so, there's about 6 months of intervening commits. It's a lot to sort through, if you think that's what needs to be done.

**Comment by dl1com** [comment]
> I expect so, there's about 6 months of intervening commits. It's a lot to sort through, if you think that's what needs to be done.

So how to proceed from here @X-sam ?
Would it make sense to re-create the PR from the current state of the 1.0.0-dev branch so we get a clean diff?

**Comment by X-sam** [comment]
Gonna re-make this PR with a fresh branch.

File: src/main/python/ayab/signal_receiver.py
Comment (X-sam) [reply src/main/python/ayab/signal_receiver.py:1]: See previous comment- we do need this as it's used by signal_sender. looks like it's treated as a registry of signals?

**Pull request closed without merging**
